### DISCUSSION TITLE
PH & ORP - Sense and Dispense

### DISCUSSION THREAD

**Discussion by SelfSufficient08** [cryptk/haomnilogic-local#57] [Ideas]
I have the Sense and Dispense system.  It would be nice to see the levels included in the integration and also the ability to turn on or off the dispense system.  
Awesome work on this.  I wasn't sure we would ever see this come out.  The primary developer of the other Hayward would probably be able to share a lot of what he has seen.  Thank you again for this !

**Comment by cryptk** [comment]
I have a actually been in contact with the developer of the other integration.  This all started when I helped them with the initial work of figuring out the local UDP API! I also have the internal API documentation courtesy of Hayward who have also been supportive of this effort (I cannot share those documents though).

We can start by having you pull the diagnostic data from the integration, this will get me some required information such as your controller configuration as well as the telemetry data from your controller, with that I can start to work on getting some support in place.

Do you have a saltwater pool with the chlorinator running in ORP mode paired up with that CSAD?

**Reply by SelfSufficient08** [reply]
Sorry for the massive delay.  Was traveling when I downloaded this and quickly set it up.  
How do I provide the two items you are requesting?  
Yes, it is a salt water pool running ORP mode paired with the CSAD.

**Comment by cryptk** [comment]
pH sensor was recently added

**Reply by SelfSufficient08** [reply]
Confirmed and works perfectly !! Thank you !!
ORP would be awesome to have some day as well..
Thank you again !

**Reply by cryptk** [reply]
ORP data is there as well as an attribute of the pH sensor.  I don't have a Sense and Dispense unit so I exposed them as a single sensor because they are exposed by the controller in a single telemetry line, but if it makes more sense to have ORP be it's own sensor, I can split that out. Just let me know which way makes more sense.

**Reply by SelfSufficient08** [reply]
Thanks for pointing that out !!  I think having it as a separate sensor would be ideal if its not too much work.  The way it works in the Hayward app is that you have the current ORP reading and then you have a slider to set the target ORP Set Point.  I have no idea if it is possible to get the ORP current reading as a sensor and then a ORP Set Point entity that allows you to choose a number from 400 to 900 (increments of 10).  That would be the ultimate setup but I have no idea how much work that actually takes to build.  
Thank you again for all of this !! Local control of the pool is amazing !  Hayward's app is so slow and sometimes just not responsive.

**Comment by spraguej248** [comment]
Ahhh, I see it now.  It's workable as is, but if it's not too much work to
split ORP to it's own sensor that would probably be better.  New users
installing this might not think to look at the ph sensor for the ORP.
Here's what it looks like for me.

[image: image.png]

Thanks!

**Reply by cryptk** [reply]
https://github.com/cryptk/haomnilogic-local/issues/106 Can you respond to that issue and attach the diagnostic data from the integration to that ticket?

By the way, I'm not sure how you are attaching pictures and such, but they aren't coming through in the GitHub UI, make sure when you respond to that issue with the diagnostic data that you do it from the GitHub UI, attachments don't tend to work if you just reply via email

**Comment by tecyah** [comment]
Should any OmniLogic ph sensor work with the integration? I have been looking into pH and ORP monitoring and possible control to add to my SWG. I also have a separate hot tub that I wouldn't mind setting up a doser on, and since I have OL automation I could just add it as another body of water.

I bought a WaterGuru but it's basically a cassette of strip tests, and it doesn't always jive with my Spin Touch. So now I'm more interested in sensor reading for pH and chlorine.

**Reply by cryptk** [reply]
as long as it's configured in the OmniLogic to where you can see the pH readings in the controller/app, it should work with the integration.
